org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/template/java/CodeSnippetTemplates.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.corext.template.java;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jface.text.templates.Template;
+
+public enum CodeSnippetTemplates {
+
+	SYSTRACE(TemplatePreferences.SYSTRACE_ID, JavaContextType.ID_STATEMENTS, TemplatePreferences.SYSTRACE_CONTENT, TemplatePreferences.SYSTRACE_DESCRIPTION),
+	FOREACH(TemplatePreferences.FOREACH_ID, JavaContextType.ID_STATEMENTS, TemplatePreferences.FOREACH_CONTENT, TemplatePreferences.FOREACH_DESCRIPTION),
```

Comment:
Should we remove the `foreach` and `fori` from the client snippets json?

Reply:
yes

Reply:
please add sysout and syserr, see if other snippets from the vscode client can be moved here

Reply:
I'm thinking whether we need to move `sysout` and `syserr` to the server side. Since they do not need to resolve any variable in the snippet template, I think leave them at the client side is a good choice (No need to let server to do the calculation).

Reply:
That'll provide snippets for more clients. And it goes along systrace

Reply:
I see.